enh: UI update of collapse description

jwallet opened the pull request:
- description in a container
- hide the container of the description.
- on hover: description has a frame around look alike file diff header to show the zone that will be collapsed
- no need to show button if no description found
- matches default bitbucket template for button light, no frame

> no description found, no bloc
![image](https://user-images.githubusercontent.com/23088305/79280569-a7896580-7e7e-11ea-9090-fefee1af229d.png)

![b](https://user-images.githubusercontent.com/23088305/79280620-c851bb00-7e7e-11ea-9316-5de9ef8f7d75.gif)


<!--
    Check those that apply, and delete the ones that don't
-->

-   [ ] I updated the CHANGELOG.md
-   [x] I tested the changes in this pull request myself
-   [x] I added Automated Tests
-   [ ] I added an Option to enable / disable this feature
-   [ ] I updated the README.md, with pictures if necessary

---

for UX purpose here the previous commit 3dafb971d16648adb1ee4f0f8b9d5555c211b4f5, this one keep a grey frame around description, like darker button, on hover it gets more opacity. but if it has no description available the grey box is still visible like currently in `dev`
![hg](https://user-images.githubusercontent.com/23088305/79348366-ebbe4980-7f02-11ea-8e97-0e5101a3aff4.gif)

jwallet committed:
ui update of collapse description

- description in a container
- hide this container
- description has a frame around look alike file diff header to show the zone that will be collapsed

reyronald reviewed (commented):
I think the hover state for the entire description is a good addition, thanks! Also good call on handling the empty description edge case 👍 

When the description is collapsed though, since there's no background or border around the caret it doesn't look like a clickable button to me, so I wonder if that could harm the user experience and discoverability 🤔

Do you think that changing that would be a bad idea?

Collapsed state **Before** (original look):

![image](https://user-images.githubusercontent.com/7514993/79553455-81c0b400-806a-11ea-8980-5e1b86666f64.png)

Collapsed state **After** (look in this PR):

![image](https://user-images.githubusercontent.com/7514993/79553469-884f2b80-806a-11ea-911a-9f496002d22c.png)

jwallet commented:
> When the description is collapsed though, since there's no background or border around the caret it doesn't look like a clickable button to me

![image](https://user-images.githubusercontent.com/23088305/79636756-01708080-8148-11ea-9ef7-1da592279921.png)


i added two different look in this pr, in the description, i think the one you are talking about is just under the `<hr>` line? it also keep the button background at all time

i changed it in the next commit because i try to make it fit with the rest of the design and i saw the comment button in files changed section looking like this.
also, the background was still appearing when the cursor was in the description zone, and i thought if someone collapse it he will know what the caret means even without background in the description

reyronald commented:
> i added two different look in this pr, in the description, i think the one you are talking about is just under the `<hr>` line? it also keep the button background at all time
> 
> i changed it in the next commit because i try to make it fit with the rest of the design and i saw the comment button in files changed section looking like this.
> also, the background was still appearing when the cursor was in the description zone, and i thought if someone collapse it he will know what the caret means even without background in the description

It's difficult to see in the second GIF because the cursor remains on top of the caret, so we don't see what happens to the background when the user hovers away, but yeah, that's what I had in mind.

I understand your reasoning for removing the background color , it's just that that way it seems like the description of the PR is the caret icon itself, and not that the caret is a button that has a toggle functionality. Does it look ugly to you that way?

jwallet commented:
no no you are right it makes sense

jwallet commented:
done

reyronald reviewed (approved):
Awesome! Thank you 🙏
